lib/Myriad/Service.pm

```diff
@@ -148,6 +181,30 @@ async method process_batch ($k, $code, $src) {
     }
 }
 
+method setup_rpc($code, $src) {
+    $src->map(async sub {
+        my $message = shift;
+
+        try {
+            my $data = await $self->$code($message->args->%*);
+            $message->response = { response => $data }
+        }
+        catch {
+            my $error = $@;
+            $message->response = { error => { code => 'internal', message => $error } };
+        };
+
+        try {
+            await $rpc->reply($message);
+        }
+        catch {
+            $log->warnf("Failed to reply to RPC message due error: %s", $@)
+        }
+
+    })->
```

Comment:
An odd linefeed here; the method name on the next line appears to have gone missing here.